icu4c/source/test/intltest/unitsdatatest.cpp

```diff
@@ -40,4 +42,80 @@ void UnitsDataTest::testGetAllConversionRates() {
     }
 }
 
+class UnitPreferencesOpenedUp : public UnitPreferences {
+  public:
+    UnitPreferencesOpenedUp(UErrorCode &status) : UnitPreferences(status) {};
+    const MaybeStackVector<UnitPreferenceMetadata> *getInternalMetadata() const { return &metadata_; }
+    const MaybeStackVector<UnitPreference> *getInternalUnitPrefs() const { return &unitPrefs_; }
+};
+
+/**
+ * This test is dependent upon CLDR Data: when the preferences change, the test
+ * may fail: see the constants for expected Max/Min unit identifiers, for US and
+ * World, and for Roads and default lengths.
+ */
+void UnitsDataTest::testGetPreferences() {
+    const char* USRoadMax = "mile";
+    const char* USRoadMin = "foot";
+    const char* USLenMax = "mile";
+    const char* USLenMin = "inch";
+    const char* WorldRoadMax = "kilometer";
+    const char* WorldRoadMin = "meter";
+    const char* WorldLenMax = "kilometer";
+    const char* WorldLenMin = "centimeter";
+    struct TestCase {
+        const char *name;
+        const char *category;
+        const char *usage;
+        const char *region;
+        const char *expectedBiggest;
+        const char *expectedSmallest;
+    } testCases[]{
+        {"US road", "length", "road", "US", USRoadMax, USRoadMin},
+        {"001 road", "length", "road", "001", WorldRoadMax, WorldRoadMin},
+        {"US lengths", "length", "default", "US", USLenMax, USLenMin},
+        {"001 lengths", "length", "default", "001", WorldLenMax, WorldLenMin},
+        {"XX road falls back to 001", "length", "road", "XX", WorldRoadMax, WorldRoadMin},
+        {"XX default falls back to 001", "length", "default", "XX", WorldLenMax, WorldLenMin},
+        {"Unknown usage US", "length", "foobar", "US", USLenMax, USLenMin},
+        {"Unknown usage 001", "length", "foobar", "XX", WorldLenMax, WorldLenMin},
+    };
+    IcuTestErrorCode status(*this, "testGetPreferences");
+    UnitPreferencesOpenedUp preferences(status);
+    auto *metadata = preferences.getInternalMetadata();
+    auto *unitPrefs = preferences.getInternalUnitPrefs();
+    assertTrue(UnicodeString("Metadata count: ") + metadata->length() + " > 200",
+               metadata->length() > 200);
+    assertTrue(UnicodeString("Preferences count: ") + unitPrefs->length() + " > 250",
+               unitPrefs->length() > 250);
+
+    // Dump all preferences... TODO: remove? This was just debugging/development output.
```

Comment:
Add the `TODO` in a separate line.

Reply:
Looks like this is resolved.  Going to mark it as such.